### ISSUE TITLE
Data provider classes

### ISSUE THREAD

**Issue by PinewoodPip** [PinewoodPip/Giickos#4]
Prerequisite: main data classes in model.

We need a system to load data. This should be done similarly to how we learnt to do it in previous subjects, via an abstract factory method, since we need to accommodate multiples sources of data:

- Mock data (for testing)
- ~~Local database (MySQL)~~ scrapped due to time constraints
- Firebase (for subscribed users; cloud storage)

There should be a single interface for loading and updating data, with classes that implement them to load the various data classes from the DBs/mock data.

At the same time, there needs to be a factory class that is injected into the model and instantiates the appropriate data providers (ex. a mock data provider for users if a mock data provider factory is injected).

For now, we need only a mock data provider. These can store information in-memory and be initialized with some placeholder data for quick testing.

**Comment by PinewoodPip** [comment]
Local MySQL database is scrapped due to time constraints.

The firebase DAO still needs some work, mainly reworking the DB structure to use documents better.

**Comment by PinewoodPip** [comment]
The firebase DB is in a good state now. The DAOs for the remaining features can have their separate issue.
